src/lib/storage/create_iterable_from_segment.hpp

```diff
@@ -30,36 +27,56 @@ class ReferenceSegmentIterable;
  * @{
  */
 
-template <typename T>
+template <typename T, bool EraseType = HYRISE_DEBUG>
 auto create_iterable_from_segment(const ValueSegment<T>& segment) {
-  return erase_type_from_iterable_if_debug(ValueSegmentIterable<T>{segment});
+  if constexpr (EraseType) {
+    return create_any_segment_iterable<T>(segment);
+  } else {
+    return ValueSegmentIterable<T>{segment};
+  }
 }
 
-template <typename T>
+template <typename T, bool EraseType = HYRISE_DEBUG>
 auto create_iterable_from_segment(const DictionarySegment<T>& segment) {
-  return erase_type_from_iterable_if_debug(DictionarySegmentIterable<T, pmr_vector<T>>{segment});
+  if constexpr (EraseType) {
+    return create_any_segment_iterable<T>(segment);
+  } else {
+    return DictionarySegmentIterable<T, pmr_vector<T>>{segment};
+  }
 }
 
-template <typename T>
+template <typename T, bool EraseType = HYRISE_DEBUG>
 auto create_iterable_from_segment(const RunLengthSegment<T>& segment) {
-  return erase_type_from_iterable_if_debug(RunLengthSegmentIterable<T>{segment});
+  if constexpr (EraseType) {
+    return create_any_segment_iterable<T>(segment);
+  } else {
+    return RunLengthSegmentIterable<T>{segment};
+  }
 }
 
-template <typename T>
+template <typename T, bool EraseType = HYRISE_DEBUG>
```

Comment:
There are two "types" being handled here, Data**Type** and Segment**Type**. As it is the latter we're erasing, I'd call this parameter "EraseSegmentType"